afs/afs-storage-api/src/test/java/com/powsybl/afs/storage/AbstractAppStorageTest.java

```diff
@@ -536,5 +535,89 @@ public void test() throws IOException, InterruptedException {
             fail();
         } catch (Exception ignored) {
         }
+
+        testUpdateNodeMetadata(rootFolderInfo);
+    }
+
+    private void testUpdateNodeMetadata(NodeInfo rootFolderInfo) throws InterruptedException {
+        NodeGenericMetadata metadata = new NodeGenericMetadata();
+        NodeInfo node = storage.createNode(rootFolderInfo.getId(), "testNode", "unkownFile", "", 0, cloneMetadata(metadata));
+        storage.flush();
+
+        checkMetadataEquality(metadata, node.getGenericMetadata());
+        discardEvents(17);
+
+        try {
+            storage.setMetadata(node.getId(), cloneMetadata(metadata));
+            storage.flush();
+            assertEventStack(new NodeMetadataUpdated(node.getId(), metadata));
+        } catch (PowsyblException e) {
+            assertThat(e).hasMessage("Not implemented");
```

Comment:
You should not consider the not implemented case as a normal case. Every storage implementation in order to pass this test has to implement the method.

Reply:
I did that because I checked the LocalAppStorage (which is not writable) and did not implement the method there. Though LocalAppStorage would fail at this test earlier anyway..